app/services/invoices/customer_usage_service.rb

```diff
@@ -165,7 +165,9 @@ def compute_amounts_with_provider_taxes
       result.fees_taxes = taxes_result.fees
 
       invoice.fees.each do |fee|
-        fee_taxes = result.fees_taxes.find { |item| item.item_id == fee.id }
+        fee_taxes = result.fees_taxes.find do |item|
+          (item.item_id == fee.item_id) && (item.amount_cents.to_i == fee.sub_total_excluding_taxes_amount_cents&.to_i)
```

Comment:
would it not be possible to generate some ids on the fees the can be used for mapping? This approach feels very fragile. 

Just using unique IDs will solve all edge cases 🤔

Reply:
Refactored 👍

Reply:
Just realised that with generating uuid on the fly, we use possibility to cache tax provider result :/ We need to use some combination of attributes on the fee if we want to keep caching available and not ping Anrok too often for current usage. So I reverted my latest commit where ID is set to `SecureRandom.uuid` on the fly
Even if current solution seems a bit hacky it covers well current usage cases + everything else should work as it is.
Let me know if you have something else in mind

Reply:
Okay, that's a good point 👍